tests/test_sign.rs

```diff
@@ -0,0 +1,66 @@
+use yaca::*;
+
+mod common;
+
+
+#[test]
+fn sign()
+{
+    // prepare:
+    let key_sym = Key::generate(&KeyType::Symmetric, &KeyLength::Bits(256)).unwrap();
+    let key_rsa_prv = Key::generate(&KeyType::Rsa(Private), &KeyLength::Bits(2048)).unwrap();
+    let key_rsa_pub = key_rsa_prv.extract_public().unwrap();
+    let hmac_simple = simple_calculate_hmac(&DigestAlgorithm::Sha512,
+                                            &key_sym, common::MSG).unwrap();
+    let cmac_simple = simple_calculate_cmac(&EncryptAlgorithm::Aes,
+                                            &key_sym, common::MSG).unwrap();
+    let sign_simple = simple_calculate_signature(&DigestAlgorithm::Sha512,
+                                                 &key_rsa_prv, common::MSG).unwrap();
+    // end prepare
+
+    let ctx = SignContext::initialize_hmac(&DigestAlgorithm::Sha512, &key_sym).unwrap();
```

Comment:
Not sure if advisable (not familiar w/ crypto testing practices) but.. Most tests omit some enum values (ex. ```DigestAlgorithm::Md5``` is not being tested). I'm not sure how much effort this would be but we could consider (for future PRs?) parameterizing some of those tests to widen the tested API surface, even if it doesn't actually increase code coverage.

Reply:
Those tests are to test the Rust wrapper only. Different DigestAlgos don't change the way wrapper behaves. They are the same in Python. Different values are tested in YACA itself (98.7% coverage).

Reply:
Assuming enums are in sync.

Reply:
I have to assume they are. This needs investigating conv functions. Doing more tests here will not be enough anyway, as sometimes those enums are interchangable and even if the conv functions were wrong such tests would not catch that. Not doing any more tests here. Not now at least.